enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -383,24 +388,43 @@ type Subnet struct {
 // +kubebuilder:validation:Pattern=`^subnet-[0-9A-Za-z]+$`
 type AWSSubnetID string
 
-// SubnetRole specifies the roles (aka functions) that the subnet will provide in the cluster.
-type SubnetRole string
+// SubnetRole specifies the role (aka functions) that the subnet will provide in the cluster.
+// +kubebuilder:validation:XValidation:rule=`self.role != 'ControlPlaneLB' || has(self.controlPlaneLB)`,message="controlPlaneLB is required when role is ControlPlaneLB"
```

Comment:
I don't think this quite works. It allows `controlPlaneLB` to be present when the role is not `ControlPlaneLB`, we normally use a ternary for this
```
self.role == 'ControlPlaneLB' ? has(self.controlPlaneLB) : !has(self.controlPlaneLB)
```